### DISCUSSION TITLE
How to remove node module resolution imports from result build?

### DISCUSSION THREAD

**Discussion by krutoo** [babel/babel#16006] [Q&A]
Hi, in my project i use babel. I want to change this:

```js
import { Foo } from '../foo'; // directory
import { Bar } from './bar'; // file without extension
```

into this:

```js
import { Foo } from '../foo/index.js';
import { Bar } from './bar.js';
```

how can i do that?

**Comment by nicolo-ribaudo** [comment]
Babel by itself does not do it, but you can use a custom plugin like the one we used internally until a few weeks ago: https://github.com/babel/babel/blob/f64b04e216be9bd34b3e7b63042b7cff1e379310/babel.config.js#L790

The reason Babel doesn't do it is that `./foo` is a prfectly valid import in browsers. If your server serves some JavaScript code at the `https://my-website.com/foo` with mime type `application/javascript`, that code will be executed. Browsers (and Deno, when running remote code) do not care about file extensions.

**Reply by krutoo** [reply]
this will work without extensions if the server is configured but will it work for child `./index.js` files?

**Reply by krutoo** [reply]
its also not working in node.js when result build has package.json with type: module
